### DISCUSSION TITLE
STATIC_DIR - Files get overwritten with default files every time container starts

### DISCUSSION THREAD

**Discussion by beastech** [open-webui/open-webui#6549] [Issues]
I want to customize the favicon and other files, but every time the container restarts it replaces my custom files with the default files.

Anybody have a fix for this, or know how to get around it?  I've tried replacing files post deployment but I would much rather use STATIC_DIR.  That is unless I am misunderstanding what this environment variable is used for.

**Comment by turgid-tributary** [comment]
I believe that on startup, if the `FRONTEND_BUILD_DIR` exists, then the files are copied from there into the static dir (I think [here](https://github.com/open-webui/open-webui/blob/1db8dec4f52fc0fa8f8f7bfbb8ea5bde41fee17d/backend/open_webui/config.py#L745-L754) ) which will cause the overwrite of your custom files.

I was able to get around this by copying my files into the build dir's static folder when building the image. Here's a really simple Dockerfile

```
FROM ghcr.io/open-webui/open-webui:main
COPY ./static /app/build/static
```

**note** Keep in mind it can be easy to violate the license when modifying branding https://docs.openwebui.com/license#open-webui-license-explained.
